src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -67,6 +74,44 @@ public void testGCSInputsAndOutputsWithSparkNio(final String gcsInput, final Str
         SamAssertionUtils.assertSamsEqual(IOUtils.getPath(outputPath), IOUtils.getPath(gcsInputPath), null);
     }
 
+    @Test(groups = "spark")
+    public void testReadAndWriteCRAMAndReferenceOnHDFS() throws Exception {
+        final GATKPathSpecifier testCram = new GATKPathSpecifier(getTestFile("count_reads.cram").getAbsolutePath());
+        final GATKPathSpecifier testRef = new GATKPathSpecifier(getTestFile("count_reads.fasta").getAbsolutePath());
+        final GATKPathSpecifier testRefDict = new GATKPathSpecifier(getTestFile("count_reads.dict").getAbsolutePath());
+        final GATKPathSpecifier testRefIndex = new GATKPathSpecifier(getTestFile("count_reads.fasta.fai").getAbsolutePath());
+
+        MiniClusterUtils.runOnIsolatedMiniCluster(cluster -> {
+            final org.apache.hadoop.fs.Path workingDirectory = MiniClusterUtils.getWorkingDir(cluster);
+
+            // copy all the test files to HDFS
+            final org.apache.hadoop.fs.Path cramHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.cram");
+            final org.apache.hadoop.fs.Path refHDFSPath = new org.apache.hadoop.fs.Path(workingDirectory, "count_reads.fasta");
+            final org.apache.hadoop.fs.Path refHDFSDictPath = new org.apache.hadoop.fs.Path(workingDirectory,"count_reads.dict");
```

Comment:
I sort of which java had lightweight type aliases so we could rename this locally to HdfsPath and avoid these nasty fully specified names...

Reply:
Yeah, I frequently wish Java had type aliases.